src/views/LoginSuccess.vue

```diff
@@ -30,226 +29,107 @@ import { Component, Emit, Vue } from 'vue-property-decorator';
-                    address,
+        const keyguardResultAccounts = this.keyguardResult.addresses.map((addressObj) => ({
+            address: new Nimiq.Address(addressObj.address).toUserFriendlyAddress(),
+            path: addressObj.keyPath,
+        }));
+
+        let tryCount = 0;
+        while (true) {
+            try {
+                tryCount += 1;
+                await WalletInfoCollector.collectWalletInfo(
+                    this.keyguardResult.keyType,
+                    this.keyguardResult.keyId,
+                    keyguardResultAccounts,
+                    (updatedWalletInfo) => this._onWalletInfoUpdate(updatedWalletInfo),
                 );
-                this.accounts.set(addressInfo.userFriendlyAddress, addressInfo);
+                this.retrievalFailed = false;
+                break;
+            } catch (e) {
+                this.retrievalFailed = true;
+                if (tryCount >= 5) throw e;
             }
-        });
-        this.accountsUpdateCount += 1; // Trigger DOM update via computed property `this.accountsArray`
-        this.contentReady = true; // The variables are set up correctly so trigger rendering the components.
-        this.storeAndUpdateResult();
-
-        if (this.keyguardResult.keyType === WalletType.BIP39) {
-            // Init Keyguard iframe to derive accounts
-            this.keyguard = new KeyguardClient();
-        }
-
-        if (this.keyguardResult.keyType === WalletType.LEDGER) {
-            if (this.keyguardResult.keyType === WalletType.LEDGER) this.walletLabel = 'Ledger Wallet';
-            // TODO: Init Ledger client to derive accounts
         }
 
-        if (this.keyguardResult.keyType === WalletType.BIP39
-         || this.keyguardResult.keyType === WalletType.LEDGER) {
-            // Init Network to check balances
-            this.network = this.$refs.network as Network;
-            await this.network.connect();
-
-            // Get balance of all accounts which were already returned from the Keyguard.
-            // FIXME: Maybe this can be included in the first iteration of `this.findAccounts()`,
-            //        but take care that at least one account is still added to the wallet,
-            //        even when all first 20 accounts have a balance of zero.
-            const userFriendlyAddresses = Array.from(this.accounts.keys());
-            const balances = await this.network.getBalances(userFriendlyAddresses);
-            userFriendlyAddresses.forEach((addr) => {
-                const addressInfo = this.accounts.get(addr);
-                addressInfo!.balance = Nimiq.Policy.coinsToSatoshis(balances.get(addr) || 0);
-                this.accounts.set(addr, addressInfo!);
-            });
-            this.accountsUpdateCount += 1;
-
-            // Kick off account detection
-            this.findAccounts();
-        }
+        // TODO network visuals with longer than 1 list of accounts during retrieval
```

Comment:
As far as I know this TODO is obsolete, as the screen will not show the account list.

Reply:
I actually do not know what is meant by this TODO

Reply:
Will be removed when that flow is reworked.